src/include/duckdb/storage/table/column_segment.hpp

```diff
@@ -76,7 +76,10 @@ class ColumnSegment : public SegmentBase {
 	void Skip(ColumnScanState &state);
 
 	// The maximum size of the buffer (in bytes)
-	idx_t SegmentSize() const;
+	inline idx_t SegmentSize() const {
```

Comment:
I'm making some hot functions inline so that they can be optimized out with the new version of D_ASSERT.